Src/FluentAssertions/Common/StringExtensions.cs

```diff
@@ -150,4 +151,105 @@ public static int CountSubstring(this string str, string substring, StringCompar
 
         return count;
     }
+
+    /// <summary>
+    /// Determines if the <paramref name="value"/> is longer than 8 characters or contains an <see cref="Environment.NewLine"/>.
+    /// </summary>
+    public static bool IsLongOrMultiline(this string value)
+    {
+        const int humanReadableLength = 8;
+        return value.Length > humanReadableLength || value.Contains(Environment.NewLine, StringComparison.Ordinal);
+    }
+
+    /// <summary>
+    /// Get the mismatch segment between <paramref name="expected"/> and <paramref name="subject"/> for long strings,
+    /// when they differ at index <paramref name="firstIndexOfMismatch"/>.
+    /// </summary>
+    public static string GetMismatchSegmentForLongStrings(this string subject, string expected, int firstIndexOfMismatch)
```

Comment:
🔧 This group of methods are not designed to be reusable and therefor should remain with the `StringEqualityStrategy`

Reply:
Moved them in [0ea0cda](https://github.com/fluentassertions/fluentassertions/pull/2307/commits/0ea0cdab575b8956d1c9f1d298951ecec216de3c).